src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -356,17 +360,17 @@ .. submodules.Select(s => s.Path).Distinct().Select(VmrPatchHandler.GetExclusion
         // We make the VMR believe it has the zero commit of the repo as it matches the dir/git state at the moment
         _dependencyTracker.UpdateDependencyVersion(new VmrDependencyUpdate(
             mapping,
-            sourceRepo.Path, // TODO = URL from BAR build
+            build.GetRepository(),
             Constants.EmptyGitObject,
             _dependencyTracker.GetDependencyVersion(mapping)!.PackageVersion,
-            Parent: null));
+            Parent: null,
+            build.AzureDevOpsBuildNumber,
+            build.Id));
 
-        IReadOnlyCollection<AdditionalRemote>? additionalRemote = build is not null
-            ? [new AdditionalRemote(mapping.Name, build.GetRepository())]
-            : [];
+        IReadOnlyCollection<AdditionalRemote>? additionalRemote = [new AdditionalRemote(mapping.Name, build.GetRepository())];
 
         string? targetVersion = null;
-        if (build?.Assets.Count > 0)
+        if (build.Assets.Count > 0)
         {
             targetVersion = build.Assets[0].Version;
         }
```

Comment:
Maybe this is better?
```suggestion
        targetVersion = build.Assets.FirstOrDefault()?.Version;
```

Reply:
it sure is shorter. Added